Fan unit instructions (#40)
uptime-industries/compute-blade: docs/docs/fan-unit/index.mdx

        light: useBaseUrl('/img/fan-unit/schematics-white-bg.svg'),
        dark: useBaseUrl('/img/fan-unit/schematics-black-bg.svg'),
    }}
    width={"75%"}
/>
<sub>*Smart Fan Unit Pictured Above*</sub>

Download the [Datasheet](/datasheet/FanUnit_datasheet_v.1.0.pdf)

## Fan Units

> The fan unit offers a compact and efficient method of cooling for two Computeblade in a BladeRunner™️ enclosure. 

Reviewer: ```suggestion
The Fan Unit offers a compact and efficient method of cooling for two Compute Blades mounted in a BladeRunner™️ enclosure. 
```

> Utilizing the 4-pin square connector on the rear of the blade to communicate with the blade. Shipped with a Noctua [NF-A4x20 5V](https://noctua.at/en/nf-a4x20-5v-pwm) a 5 Volt PWM fan.

Reviewer: ```suggestion
The fan unit uses a 4-pin square connector to communicate with the Compute Blade. It is also shipped with a 5V PWM fan, [Noctua NF-A4x20 5V](https://noctua.at/en/nf-a4x20-5v-pwm).
```

> Their are two types of Fan Units, Standard and Smart.

Reviewer: ```suggestion
There are two types of Fan Units, **Standard** and **Smart**.
```

### Standard

> The Standard Fan Unit offers basic PWM Functionality but a no fuss setup requiring now out of box configurations to get going.

Reviewer: ```suggestion
The Standard Fan Unit offers basic PWM functionality, and allows for a no-fuss setup with zero configuration required.
```

> With that simplicity comes a drawback, the Standard Fan Unit can not take PWM Signals from both blades.

Reviewer: ```suggestion
Because of the simplicity in its design, the Standard Fan Unit is unable to receive PWM Signals from both Compute Blades.
```

The PWM signals can only be sent by the blade on the J1 header (Power Input 1 in the above picture)

:::info

>     The fan tach signal can be read by both blades see the [GPIO guide](#blade-connector) for the pin out

Reviewer: ```suggestion
    The fan tachometer signal can be read by both blades. See the [GPIO guide](#blade-connector) for the pin out configuration.
```

:::

### Smart

To compliment the drawback of the Standard Fan Unit, the Smart Fan Unit uses the [Raspberry Pi RP2040](https://www.raspberrypi.com/products/rp2040/specifications/).
Utilizing the 32-bit dual [ARM Cortex-M0+](https://developer.arm.com/Processors/Cortex-M0-Plus) microcontroller. The Smart Fan Unit is capable of full bidirectional communication to both of the cooled blades, programmable status LEDs, and sensors. 

> This increased functionality comes with the down side of increased complexity in the setup. Requiring the setup of the Smart Fan Unit and The computeblade.

Reviewer: ```suggestion
This enhanced functionality comes with more complexity in its setup and configuration. It requires configuration of both the Smart Fan Unit and the Compute Blade.
```

> ## Which Fan Unit should you uses

Reviewer: ```suggestion
## Which Fan Unit Should You Use?
```

> If simple compute task are your intended work load or you are using the blade in a hobbyist environment the Standard Fan Unit may be right for you.

Reviewer: ```suggestion
If simple compute tasks are your intended workload or you are using the blade in a hobbyist environment, the Standard Fan Unit is likely right for you.
```

> If you are looking for detailed information about the state of the fan or have more dynamic work loads the SMat Fan Unit may be for you.

Reviewer: ```suggestion
If you are looking for detailed information about the state of the fan, or have more dynamic workloads the Smart Fan Unit may be necessary.
```

        </table>
    </TabItem>
</Tabs>
For additional information on the NF-A4x20 5V, see the [datasheet](https://noctua.at/pub/media/blfa_files/infosheet/noctua_nf_a4x20_5v_datasheet_en_web_1.pdf)
## Block Scheme

<Tabs groupId="fanunit" defaultValue="smart">
    <TabItem value="smart" label="Smart">
        <img src="/img/fan-unit/smart-block.svg" alt="Standard Fan Unit Block Scheme" width={'50%'}/>
    </TabItem>
    <TabItem value="sta" label="Standard">
        <img src="/img/fan-unit/sta-block.svg" alt="Standard Fan Unit Block Scheme" width={'70%'}/>
    </TabItem>
</Tabs>

## GPIO

### Blade Connector

:::danger

> The Blade connector Is a square un-keyed connector. This means that is is physically possible to misaligned the connector

Reviewer: ```suggestion
The Blade connector is a square un-keyed connector. This means that is is physically possible to misalign the connector.
```

:::

<Tabs groupId="fanunit" defaultValue="smart">
    <TabItem value="smart" label="Smart">
        <img src='/img/fan-unit/smart-gpio.svg' width={'70%'}/>
    </TabItem>
    <TabItem value="sta" label="Standard">
        <img src='/img/fan-unit/sta-gpio.svg' width={'70%'}/>
    </TabItem>
</Tabs>

### Fan Header

> The header uses a four-pin Molex connector(KK family). And follows the [Noctua v5 Fan pin configuration](https://faqs.noctua.at/en/support/solutions/articles/101000081757-what-pin-configuration-do-noctua-fans-use-).

Reviewer: ```suggestion
The header uses a four-pin Molex connector([KK family](https://www.molex.com/en-us/products/connectors/wire-to-board-connectors/kk-connectors)). And follows the [Noctua 5V Fan pin configuration](https://faqs.noctua.at/en/support/solutions/articles/101000081757-what-pin-configuration-do-noctua-fans-use-).
```

<img src='/img/fan-unit/fan-header.svg' width={'70%'}/>

> ### Debug (Smart Fan Unit Only)

Reviewer: ```suggestion
### Debugging (Smart Fan Unit Only)
```

> The J5 debug header enables debug of the RP2040. Using tool like the [Raspberry Pi Debug Probe](https://www.raspberrypi.com/products/debug-probe/)

Reviewer: ```suggestion
The J5 debug header enables debugging of the RP2040 via a tool like the [Raspberry Pi Debug Probe](https://www.raspberrypi.com/products/debug-probe/).
```
